src/linear_algebra/basic.lean

```diff
@@ -1641,6 +1641,19 @@ rfl
 
 end prod
 
+section curry
+
+variables (V V₂ R)
+
+/-- Linear equivalence between an uncurried and curried function. -/
+protected def curry :
+  (V → V₂ → R) ≃ₗ[R] (V × V₂ → R) :=
+{ add := λ _ _, by ext z; cases z; refl,
```

Comment:
The preferred style is now not to use `;` except where it is necessary --- that is, when you want to run one tactic on multiple produced goals.

Instead use `by { ext z, cases z, refl, }`, with braces.

(Lots of older code in mathlib was written before this style rule was introduced, and never updated.)